slidez-core/src/main/java/com/binarystudio/academy/slidez/domain/user/UserService.java

```diff
@@ -16,64 +17,84 @@
 @Service
 public class UserService {
 
-	@Autowired
-	protected PasswordEncoder passwordEncoder;
-
-	@Autowired
-	private UserRepository userRepository;
-
-	@Autowired
-	private AuthService authService;
-
-	public boolean isEmailPresent(String email) {
-		return this.userRepository.existsByEmail(email);
-	}
-
-	public Optional<User> findByEmail(String email) {
-		return this.userRepository.findByEmail(email);
-	}
-
-	public Optional<User> findByToken(String token) {
-		String email = this.authService.getLoginFromToken(token);
-		if (email == null) {
-			return Optional.empty();
-		}
-		return findByEmail(email);
-	}
-
-	public Optional<UserDetailsDto> getDetailsByToken(String token) {
-		String email = this.authService.getLoginFromToken(token);
-		if (email == null) {
-			return Optional.empty();
-		}
-		Optional<User> userOptional = findByEmail(email);
-		User user = userOptional.orElseThrow();
-		UserDetailsDto userDetailsDto = UserMapper.INSTANCE.mapUserToUserDetailsDto(user);
-		return Optional.of(userDetailsDto);
-	}
-
-	public User create(UserDto userDto) {
-		User user = UserMapper.INSTANCE.userDtoToUser(userDto);
-		user.setPassword(this.passwordEncoder.encode(userDto.getPassword()));
-		return this.userRepository.saveAndFlush(user);
-	}
-
-	public Optional<User> getById(UUID id) {
-		return this.userRepository.findById(id);
-	}
-
-	public User createByEmail(String email) {
-		User user = new User();
-		user.setEmail(email);
-		user.setPassword(this.passwordEncoder.encode(""));
-		return this.userRepository.save(user);
-	}
+    @Autowired
+    protected PasswordEncoder passwordEncoder;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private AuthService authService;
```

Comment:
Field injection is not recommended, but it's no big deal. But you have here circular dependency - userService -> authService -> userService, and that's not good. You should first merge with develop - I replaced authService with jwtProvider to get rid of circular dependency, but that's not good either - UserService mustn't know about our security policy - hence such methods as getSomethingFromToken should be moved to authService (Or removed completely - extract principal and just get by email)